### DISCUSSION TITLE
Charging Stations -- for cars? for bikes? quest for it?

### DISCUSSION THREAD

**Discussion by hip365** [streetcomplete/StreetComplete#5782] [General]
Hi, 
first of all; I'm pretty new to streetcomplete (and even openstreetmap), if I get something wrong, bear with me, I'll try to catch up.

I'm just gathering information on charging stations, especially for e-bikes, in the northern rhein-main-area in Germany. While correcting and completing data on some nodes, I realized, data quality seems to have a lot of room for improvement, to be polite.

In the state of Hessen, there are close to two-thousand charging stations in OSM. That's pretty good. But, nearly two-thirds of these have no indication, whether they're for cars, bikes or both.

In OSM-terms: most of the "amenity=charging_station" nodes have neither "bicycle" nor "motorcar" tags with them.

To play with numbers, here's my query-scratchpad:
	https://overpass-turbo.eu/s/1P9R

Wouldn't it make sense to have quests in SC to qualify the type of charging station?
Like "Which types of vehicles can be charged here?" or something similar?

Or ask for the available plugs and outlets (schuko-outlet, plugs for Bosch, Shimano, Yamaha, etc.)?

**Comment by mnalis** [comment]
Yes, it would.. 

- But there are problems, see https://github.com/streetcomplete/StreetComplete/issues/4003 and https://github.com/streetcomplete/StreetComplete/issues/5164

**Reply by hip365** [reply]
Thanks for pointing me to these two (open) issues.

It seems, we're entangled in the complexity of the muddy world outside.

With "amenity=charging_station" OSM tries to map out several dimensions of information (not exhaustive):
- user group: bike/car driver, etc.
- technical criteria: available sockets, plugs, wattage, ...
- legal issues: access allowed for whom by law or operator
- contract stuff: fees, payment methods, customer cards, ...
- physical issues: parking lot large enough for heavy trucks?
...and some more.

These dimensions are not independant. It's easy to create datasets with contradictions and there's a plethora of edge and corner cases.

A friend of mine just got a charging station at his office. It's a "two cars and four bikes"-combo. 
What's the correct(?) value for the "capacity"-key?

Trying to cover all this is heroic -- we don't dare to start, because there are details down the road, that are not claryfied yet. Instead, we limit the usefulness of the OSM-data for the vast majority of users.

How do you eat an elephant? Let's start small and keep it simple.

Here's my usecase:
> As an e-biker I want to be able to query the locations of charging stations nearby / along my route, in order to juice up my batteries (or plan my trip accordingly).

Technically, setting "bicycle=yes|no" would benefit a lot, as this is the main filter criteria for end-user applications (komoot, mapy, ...) to display charging stations for bikes.

A quest like "Can bikes be charged here?" "Yes/No/Don't know" would do the job.

Any objections to start with a quest simple as that?

**Reply by mnalis** [reply]
> A friend of mine just got a charging station at his office. It's a "two cars and four bikes"-combo.
What's the correct(?) value for the "capacity"-key?

It depends on the specific model. If all 6 vehicles can be charged at the same time (which is probably not the case) it would be `amenity=charging_station` + `capacity=6`. You would also add `socket:chademo=2` + `socket:bosch_3pin=4` or whatever combination of sockets is actually available, to make it actually useful. And if even more complexity is needed, you'd make that `amenity=charging_station` an area, and put multiple `man_made=charge_point` POIs with their own tags inside it.

> A quest like "Can bikes be charged here?" "Yes/No/Don't know" would do the job.

What would you expect SC to tag for each of the answers, and how would correct answers be explained to the user? (i.e. what does "yes" mean? All types of e-bikes can be charged there? Or at least one type can be charged? Must you bring your own charging brick?)

> Any objections to start with a quest simple as that?

I'm afraid it would be feel-good but completely useless information (see other big post for more detailed explanation why)

**Comment by mnalis** [comment]
**TL;DR:** Only way that this information is going to be useful IMHO is if it precise enough, i.e. if it specify actual sockets available. The rest of this post is detailed explanation why.

---

> Technically, setting "bicycle=yes|no" would benefit a lot, as this is the main filter criteria for end-user applications (komoot, mapy, ...) to display charging stations for bikes.

There is very strong OSM popular opinion that `bicycle=yes|no` is ***exclusively*** about ***legal*** access rights, not about any fitness for purpose. (There is also the issue if the charging station also included e.g. air compressor whether the `bicycle=yes` no applies to one, the other, or both.) 
<details>
<summary>Also, bicycle is incorrect...</summary>
And, [electric_bicycle=*](https://wiki.openstreetmap.org/wiki/Key:electric_bicycle) would probably be better then `bicycle=*` anyway, as you can hardly "charge" regular bicycle. (then there are [speed_pedelec=*](https://wiki.openstreetmap.org/wiki/Key:speed_pedelec) and other electric mofas/mopeds/motorcycles, which are not legally bicycles; what about them?)
</details>

Correct tag would probably instead be [service:bicycle:charging=*](https://wiki.openstreetmap.org/wiki/Key:service:bicycle:charging)

But **most importantly** IMHO, while `service:bicycle:charging=no` might be clear (i.e. no bicycle ever may be able to charge here), `service:bicycle:charging=yes` is more problematic. Does it mean **all** types of electric bicycles can be charged here? Or **at least one** type? Or some kind of majority (simple majority, two-thirds majority, 95% majority. And how one counts/estimates that) ?

People want to know if **their** bike can be charged there. Imagine if it was extremely popular that fuel stations have **only** gasoline or **only** diesel. `motorcar=yes` would be pretty useless, you'd absolutely want `fuel:diesel=yes`, `fuel:octane98=yes` etc. instead. Luckily for ICE car owners, vast majority of fuel stations offer both. Not so lucky for electric bicycle (or even electric car!) owners. 


In this little part of Europe for example, popular bicycle charger sockets are either  `shimano_steps_5pin` or `bosch_3pin` or `Schuko`. And very rarely are all 3 available at the same time. (And occasionally other types popup too, like `bosch_5pin` etc)

While first two (Shimano, Bosch) are great if I have bicycle compatible with them, they are completely useless if my bicycle has other charging system. `Schuko` is more universal (if you are a local; i.e. not a tourist from country which uses other "default" socket standard!), but it requires that you _carry your own charger with you_ at all times (which many locals won't carry unless planing longer multi-days trips in advance). So again, no good generic _"yes you can charge bicycle here"_ answers.

So, many people would be quite disappointed (more likely outright angry) if they go to generically tagged `service:bicycle:charging=yes` station, and found out that it only has (for example) `socket:ropd=yes` (which their bike doesn't support).

On the other hand, if you expect that people will carry their own charger to plug in (e.g. to use `Schuko` AKA `IEC type F`, or `type E` etc.) , and you want to mark that _"at least some bicycles can be charged here"_, then you would end up in situation where all car charges should be marked as bicycle-compatible, because of existence of [J1772 to NEMA 5-15/5-20 EV Charger Adapter](https://forums.electricbikereview.com/threads/charging-ebikes-on-ev-charging-stations.50229/) for CCS Combo 1 and similar adapters allowing electric bicycles to use _"car"_ charges.

So, yeah, it is complicated, and **complicated in a way that _requires_ being specific** (i.e. specifying [socket=*](https://wiki.openstreetmap.org/wiki/Key:socket:*)) **to be useful _at all_**.
